pkg/vault/vault.go

```diff
@@ -53,6 +53,98 @@ func GetSecrets(ctx context.Context, secretValues *map[string]map[string]string,
 	return nil
 }
 
+// CreateSecret takes a given key for an engine, and adds a new key/value pair in vault.
+func CreateSecret(ctx context.Context, engine, key, value string) error {
+	environment := os.Getenv("ENVIRONMENT")
+
+	if environment == "production" {
+		log.SetFormatter(&log.JSONFormatter{})
+		log.SetLevel(log.WarnLevel)
+	} else {
+		log.SetFormatter(&log.TextFormatter{})
+		log.SetLevel(log.TraceLevel)
+	}
+
+	config, err := loadVaultEnvironment()
+	if err != nil {
+		return fmt.Errorf("load client environment: %w", err)
+	}
+
+	vc, err := NewVaultClient(ctx, config)
+	if err != nil {
+		return fmt.Errorf("error initializing vault client: %w", err)
+	}
```

Comment:
This is not DRY with the two functions below.  Can we clean up this repetition?

Reply:
I am pretty close to removing logging from here entirely which would help clean it up even more.